src/login/logind-dbus.c

```diff
@@ -2323,20 +2323,49 @@ static int method_schedule_shutdown(sd_bus_message *message, void *userdata, sd_
 
 static int method_cancel_scheduled_shutdown(sd_bus_message *message, void *userdata, sd_bus_error *error) {
         Manager *m = userdata;
+        const char *action;
         bool cancelled;
+        int r;
 
         assert(m);
         assert(message);
 
         cancelled = m->scheduled_shutdown_type != NULL;
+
+        if (!cancelled)
+                goto done;
```

Comment:
goto is useful, but I'd always avoid it, hence maybe write this as:

```c
if (!m->scheduled_shutdown_type)
        return sd_bus_reply_method_return(message, "b", false);
```

and then change the last line to:

```c
return sd_bus_reply_method_return(message, "b", true);
```

Reply:
will do